src/cloud-providers/ibmcloud-powervs/provider.go

```diff
@@ -54,6 +69,44 @@ func (p *ibmcloudPowerVSProvider) CreateInstance(ctx context.Context, podName, s
 		return nil, err
 	}
 
+	if len(*p.serviceConfig.PreCreatedInstances) > 0 {
+		var conn net.Conn
+		instance := (*p.serviceConfig.PreCreatedInstances)[0]
+		*p.serviceConfig.PreCreatedInstances = (*p.serviceConfig.PreCreatedInstances)[1:]
+		logger.Printf("Using instance from pre-created pod VM pool, name: %s ,id:%s", instance.Name, instance.ID)
+
+		address := fmt.Sprintf("%s:%s", instance.IPs[0].String(), p.serviceConfig.PudPort)
+		logger.Printf("Connecting to pre-created VM: %s", address)
+		ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+		defer cancel()
+
+		err := retry.Do(
+			func() error {
+				var err error
+				conn, err = (&net.Dialer{}).DialContext(ctx, "tcp", address)
+				return err
+			},
+			retry.Attempts(0),
+			retry.Context(ctx),
+			retry.MaxDelay(5*time.Second),
+		)
+		if err != nil {
+			err = fmt.Errorf("failed to establish connection to process-user-data: %s: %w", address, err)
+			logger.Print(err)
+			return nil, err
+		}
+		defer conn.Close()
+
+		logger.Println("Connection established, prepare to send data..")
+		_, err = conn.Write([]byte(userData))
+		if err != nil {
+			return nil, err
+		}
+
+		logger.Printf("Successfully sent user-data to the pre-created VM %s, IP:%s", instance.Name, instance.IPs[0])
```

Comment:
Does that cause the pod reinitialization in the pod vm image? Or is that something that needs to be done elsewhere?